src/os/bluestore/BlueStore.cc

```diff
@@ -6046,14 +6046,13 @@ void BlueStore::_txc_state_proc(TransContext *txc)
         sb->bc.finish_write(txc->seq);
       }
       txc->shared_blobs_written.clear();
+      _txc_finalize_kv(txc, txc->t);
```

Comment:
This change breaks if you are using ExtentFreelistManager.

We should add an additional virtual method to FreelistManager.h called something bool supports_parallel_transactions() that is true for the bitmap version and false for the extent version.  Then we drop this change, and add a new patch that prevents you from using sync_transaction or sync_submit_transaction entirely.  (We could try to make sync_submit_transaction work but I don't think there's any reason to bother.)